networksimulator.py

```diff
@@ -189,22 +198,29 @@ def run(self, duration=sys.float_info.max):
         """Runs the simulation for the given duration.
         
         Args:
-            duration (float): the duration of the simulation in milliseconds.
+            duration (float): the duration of the simulation in seconds.
                 By default, the simulation runs until termination.
 
         """
-        pass
+        while self.pq and self.num_active_flows > 0 and self.cur_time < duration:
```

Comment:
i don't really remember how conditions work, but i'm not convinced that this is right